### PR TITLE
Prevent duplicate scores in XML input

### PR DESCRIPTION
**Pull request by paulromano**
This pull request prevents duplicate scores in the XML input, as discussed with @samuelshaner and @wbinventor. It was a little tricky since some scores have multiple representations, for example, the following would gives the same score three times:
```xml
<scores>16 n2n (n,2n)</scores>
```
So, in general I couldn't just compare strings. Instead, I compare (score, moment order) for each bin. The complication there is that Pn and Yn expansion scores have multiple bins that store the same score and order, so for those, I calculate the number of bins and just "skip-ahead" so that they're not compared.

This PR also improves the documentation on available score types, explicitly listing a bunch of reactions that users may not be aware of, e.g. `<scores>(n,gamma) (n,p) (n,npa)</scores>`. Before, you'd have to know the MT number for the reactions. Now, the our documentation tells you everything you need. I also turned the list of scores into a table so it is formatted better (note I had to do a CSS hack to get the table to wrap in the HTML output).

### CODE REVIEW

File: docs/source/usersguide/input.rst [hunk 1]
```diff
@@ -1475,114 +1475,233 @@ The ``<tally>`` element accepts the following sub-elements:
...
-      Number of scoring events. Units are events per source particle.
+    A space-separated list of the desired responses to be accumulated. The accepted
+    options are listed in the following table:
+
+    .. table:: Score types available in OpenMC
+
+        +----------------------+---------------------------------------------------+
+        |Score                 | Description                                       |
+        +======================+===================================================+
+        |flux                  |Total flux in particle-cm per source particle.     |
```
Comment (wbinventor) [inline docs/source/usersguide/input.rst]: do we need the note about the analog estimator being equal to the collision estimator for the flux?

Comment (paulromano) [inline docs/source/usersguide/input.rst]: It's actually below the table now. Not the most obvious place ever, but I couldn't figure out how to stick a note directive inside a table cell.

File: docs/source/usersguide/input.rst [hunk 2]
```diff
@@ -1475,114 +1475,233 @@ The ``<tally>`` element accepts the following sub-elements:
...
-
-    :events:
-      Number of scoring events. Units are events per source particle.
+    A space-separated list of the desired responses to be accumulated. The accepted
+    options are listed in the following table:
+
+    .. table:: Score types available in OpenMC
+
+        +----------------------+---------------------------------------------------+
+        |Score                 | Description                                       |
```
Comment (wbinventor) [inline docs/source/usersguide/input.rst]: Two thoughts on this table:

* perhaps put the scores in double quotations to mirror their appearance in either XML or Python
* perhaps alphabetically order the scores?

I realize that these are separate from the motivation behind this PR, but are two things that come to mind when reading this table over.

File: docs/source/usersguide/input.rst (hunk 1 above)
Comment (wbinventor) [inline docs/source/usersguide/input.rst]: Oh ok, I hadn't yet reached that far!

File: docs/source/usersguide/input.rst (hunk 2 above)
Comment (wbinventor) [inline docs/source/usersguide/input.rst]: A final thought is that there is a lot of somewhat redundant "Units are reactions per source particle" in the table. What are your thoughts on creating sub-sections in the table for the following:

* Units are Reactions per Source Particle
* Units are Neutrons Produced per Source Particle
* Other (MeV per Source Particle, Centimeters per Second, etc.)

File: src/state_point.F90 [hunk 3]
```diff
@@ -310,54 +310,7 @@ subroutine write_state_point()
           call write_dataset(tally_group, "n_score_bins", tally%n_score_bins)
           allocate(str_array(size(tally%score_bins)))
           do j = 1, size(tally%score_bins)
-            select case(tally%score_bins(j))
```
Comment (wbinventor) [inline src/state_point.F90]: It's always nice to shrink the size of "state_point.F90" and "summary.F90!"

**Comment by wbinventor** [comment]
This looks fine to me save for my questions about the new table of scores. Also, GitHub claims this has merge conflicts with "develop" which must be resolved before merging.

**Comment by samuelshaner** [comment]
Thanks for addressing the duplicate scores issue @paulromano! As you described, the behavior for the two duplicate score cases is:

* If two of the same scores exist (e.g. n2n and 16), error out and inform the user of duplicate scores.
* If two scores have duplicate score bins (e.g. scatter-P3 and scatter-P4), run simulation (even though there will be identical score bins).

The issue with the second case is that the Python API will read in the tally data and still error out due to duplicate scores since the Python API breaks up the scatter-PN into (N+1) different scores. I was looking at `input_xml.F90` and removing the duplicate bins appears to be challenging. I believe it would require creating new `tally % score_bins()` and `tally % moment_order()` arrays and leaving out any duplicate bins. Are we okay with this behavior since the use cases are probably fairly uncommon? Should the fortran code be modified to remove duplicate score bins? Or the Python API modified to ignore duplicate scores (I don't think this is as easy as it sounds)? I feel like the use cases are probably fairly uncommon and an easy work around that would allow the tally data to be imported into the Python API is just to create another tally, so I'm fine with the Python API just erring out and not supporting this case. What are your thoughts?

File: docs/source/usersguide/input.rst [hunk 4]
```diff
@@ -1475,114 +1475,233 @@ The ``<tally>`` element accepts the following sub-elements:
...
-      .. note::
-         The ``analog`` estimator is actually identical to the ``collision``
-         estimator for the inverse-velocity score.
-
-    :events:
-      Number of scoring events. Units are events per source particle.
+    A space-separated list of the desired responses to be accumulated. The accepted
+    options are listed in the following table:
+
+    .. table:: Score types available in OpenMC
```
Comment (samuelshaner) [inline docs/source/usersguide/input.rst]: I'd recommend **emphasizing** to make it stand out as it looks a little awkward as normal text.

**Comment by paulromano** [comment]
@wbinventor Your comments have been addressed.

@samuelshaner Yeah, it's not simple to completely fix that problem. I'm ok just ignoring it for now. If we want to have the Python API just error out, that should be fine.

**Comment by wbinventor** [comment]
Looks great, thanks for the revisions @paulromano! The duplicate score err'ing out can be iterated on in future PRs if we ever deem it a real problem.

**Pull request merged**